src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
 import { GitHubProvider } from "../../helper/gitHubHelper";
 import { SodiumLibHelper } from "../../helper/sodium/SodiumLibHelper";
+import { GitHubOrganization, GitHubRepo } from '../../model/models';
 import { RestClient } from "../restClient";
+const uuid = require('uuid/v4');
 
 const UserAgent = "deploy-to-azure-vscode";
 
 export class GithubClient {
 
     private patToken: string;
     private url: string;
+    private listOrgPromise: Promise<void | GitHubOrganization[]>;
+
 
     constructor(patToken: string, remoteUrl: string) {
         this.patToken = patToken;
         this.url = remoteUrl;
+        this.listOrgPromise = this.listOrganizations(this.patToken);
     }
 
-    public async createOrUpdateGithubSecret(secretName: string, body: string): Promise<void> {
-        let secretKeyObject: GitHubSecretKey = await this._getGitHubSecretKey();
-        let sodiumObj = new SodiumLibHelper(secretKeyObject.key);
-        let encryptedBytes: Uint8Array = sodiumObj.encrypt(body);
-        let encryptedBytesAsString: string = SodiumLibHelper.convertUint8ArrayToString(encryptedBytes);
-        let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
-        await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+    public createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+        let restClient =new RestClient();
+        let Url = "https://api.github.com/orgs/" + orgName + "/repos";
+        try{
+            return restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: Url,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + this.patToken,
+                    "User-Agent": UserAgent
+                },
+                method: 'POST',
+                body: {
+                    "name": repoName,
+                    "description": "Repo created from VScode",
+                    "homepage": "https://github.com",
+                    "private": false,
+                    "has_issues": true,
+                    "has_projects": true,
+                    "has_wiki": true
+                },
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((detail: GitHubRepo) => {
+                    return detail;
+                }).catch(error=>{
+                        return null;
+                });
+        }
+        catch(error){
+            return null;
+        }
     }
 
-    private async _getGitHubSecretKey() : Promise<GitHubSecretKey> {
-        let request = <UrlBasedRequestPrepareOptions> {
-                url: GitHubProvider.getFormattedGitHubApiUrlBase(this.url) + "/actions/secrets/public-key",
+    public async listOrganizations(PATtoken: string, forceRefresh?: boolean): Promise<void | GitHubOrganization[]> {
```

Comment:
PAT token has already been set in the ctor. It should not be needed here.

Reply:
Okay, will rectify it.